### PR TITLE
EventManager. Once again

### PR DESCRIPTION
**Pull request by alexeevdv**

### CODE REVIEW

File: src/Game/CritterObject.cpp [hunk 1]
```diff
@@ -482,8 +485,8 @@ void GameCritterObject::think()
             _orientation = hexagon()->orientationTo(movementQueue()->back());
             auto animation = _generateMovementAnimation();
             animation->setActionFrame(_running ? 2 : 4);
-            animation->addEventHandler("actionFrame",    std::bind(&GameCritterObject::onMovementAnimationEnded, this, std::placeholders::_1));
-            animation->addEventHandler("animationEnded", std::bind(&GameCritterObject::onMovementAnimationEnded, this, std::placeholders::_1));
+            EventManager::getInstance()->addHandler("actionFrame",    std::bind(&GameCritterObject::onMovementAnimationEnded, this, _1), animation);
+            EventManager::getInstance()->addHandler("animationEnded", std::bind(&GameCritterObject::onMovementAnimationEnded, this, _1), animation);
```
Comment (phobos2077) [inline src/Game/CritterObject.cpp]: This is procedural programming... it's BAD. Why don't stick with OOP approach (leaving addEventHandler method for objects)?

File: src/Game/Game.cpp [hunk 2]
```diff
@@ -155,7 +156,8 @@ void Game::popState()
     _statesForDelete.push_back(state);
 
     auto event = new StateEvent("deactivate");
```
Comment (phobos2077) [inline src/Game/Game.cpp]: do we need new/delete here? We don't just use normal object in a tight scope?

File: src/Game/Object.cpp [hunk 3]
```diff
@@ -43,19 +45,22 @@
...
+GameObject::GameObject() : EventSender()
 {
 }
 
 GameObject::~GameObject()
 {
     delete _ui;
-    delete _floatMessage;
+    delete _floatMessage;    
+    EventManager::getInstance()->removeHandlers(this);
```
Comment (phobos2077) [inline src/Game/Object.cpp]: Why don't move this to EventSender class virtual destructor?

File: src/Graphics/Renderer.cpp [hunk 4]
```diff
@@ -176,7 +177,7 @@ void Renderer::think()
             _fadeDone = true;
 
             auto event = new StateEvent("fadedone");
-            Game::getInstance()->states()->back()->emitEvent(event);
+            EventManager::getInstance()->handle(event);
             delete event;
             return;
```
Comment (phobos2077) [inline src/Graphics/Renderer.cpp]: You can enclose in scope instead of new/delete.

File: src/Event/EventManager.cpp [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+EventManager* EventManager::_instance = nullptr;
+
+EventManager::EventManager()
+{
+}
+
+EventManager* EventManager::getInstance()
+{
+    if(!_instance)
```
Comment (phobos2077) [inline src/Event/EventManager.cpp]: Missed space ;-) and look at "for" loops below

File: src/Game/Object.cpp (hunk 3 above)
Comment (alexeevdv) [inline src/Game/Object.cpp]: Excellent idea!

**Pull request closed without merging**